Autopilot/PathManager/Src/landingTakeoffManager.cpp

```diff
@@ -0,0 +1,202 @@
+#include "landingTakeoffManager.hpp"
+#include <math.h>
+
+double LandingManager::changingAltitude(const SFOutput_t * input, const _PathData * aimingPoint, const _PathData * intersectionPoint, const _PathData * stoppingPoint)
+{
+    //vector declaration
+    Vector3D aPoint(aimingPoint->longitude, aimingPoint->latitude, aimingPoint->altitude); //aiming point
+    Vector3D iPoint(intersectionPoint->longitude, intersectionPoint->latitude, intersectionPoint->altitude); //intersection point
+    Vector3D sPoint(stoppingPoint->longitude, stoppingPoint->latitude, stoppingPoint->altitude); //stopping point
+    Vector3D cPoint(input->longitude, input->latitude, input->altitude); //current point
+    Vector3D projectedPoint;
+
+    //calculating the vectors
+    Vector3D vectorAI(iPoint.x - aPoint.x, iPoint.y - aPoint.y, iPoint.z - aPoint.z); //aiming point --> intersection point
+    Vector3D vectorAS(sPoint.x - aPoint.x, sPoint.y - aPoint.y, sPoint.z - aPoint.z); //aiming point --> stopping point
+
+    Vector3D vectorAC(cPoint.x - aPoint.x, cPoint.y - aPoint.y, cPoint.z - aPoint.z); //aiming point --> current point
+    Vector3D normalVector;
+    
+    //normal of the plane created by aiming, stopping, and intersection point
+    normalVector = vectorAI.crossProduct(vectorAS);
+
+    //projecting vectorAC to the normal
+    //equation is the dot product of normal and vectorAC, divided by the norm of the normal squared, multiplied by the normal vector
+    Vector3D projectionACToNormal;
+
+    projectionACToNormal = normalVector * (normalVector.dotProduct(vectorAC) / pow(normalVector.norm(),2));
+
+    //project point onto plane (cpoint minus the projection)
+
+    projectedPoint = cPoint - projectionACToNormal;
+
+    //determine the equation of the line of the slope and sub in projected values (symmetric form)
+    double altitude = (((projectedPoint.x - aPoint.x) / vectorAI.x) + aPoint.z) * vectorAI.z;
+
+    return altitude;
+}
+
+double LandingManager::approachSpeed(const bool *ifPackage)
+{
+    //approach speed calculation for both package and no package scenarios
+    if(*ifPackage)
+    {
+        return STALL_SPEED_WITH_PACKAGE * 1.3;
+    }
+    else 
+    {
+        return STALL_SPEED_NO_PACKAGE * 1.3;
+    }
+}
+
+double LandingManager::slowFlightSpeed(const bool *ifPackage)
+{
+    //slow flight speed calculation for both package and no package scenarios
+    if(*ifPackage)
+    {
+        return STALL_SPEED_WITH_PACKAGE + 2;
+    }
+    else 
+    {
+        return STALL_SPEED_NO_PACKAGE + 2;
+    }
+}
+
+_LandingPath LandingManager::createSlopeWaypoints(const Telemetry_PIGO_t * input, const float * currentAltitude)
```

Comment:
THere is a c++y way of passing these variables by reference. What you've got here is the old C way, which I started building into our code cuz I didn't know any better. Here's the right way to do it in C++.
https://www.ibm.com/docs/en/zos/2.2.0?topic=calls-pass-by-reference-c-only

The advantage to this is you don't ever have to worry about pointer de referencing or people passing in trash pointers ever again.